### ISSUE TITLE
Configuration: logging options

### ISSUE THREAD

**Issue by NicolasDP** [input-output-hk/jormungandr#65]
We need to extend the logging option in the configuration file:

* [x] where to dump the logs;
* [x] the verbosity level of the logs;

I can see cases where we might want to have the logs redirected to syslog but also the warning and the error ones to be on the standard error output too.

might need to sync with #55

**Comment by NicolasDP** [comment]
we need to finish the _where to output the logs_ part

**Comment by CodeSandwich** [comment]
@NicolasDP Here's my propositon:
# Current design
## User input
- CMD provides verbosity in form of `-vvv` (default 0 a.k.a. `info`) and optional log format
- Config provides optional verbosity and optional log format
## Logger setup
- If config provides verbosity, it overwrites CMD verbosity
- If CMD does not provide log format, it's set to `plain`
- Logger uses CMD verbosity and log format
## Consequences
- Only 1 logging output
- Config log format is always ignored

# Design proposition
## User input
- CMD provides verbosity in form of `-vvv` (default 0 a.k.a. info) and optional log format
- Config provides **list** of structures with mandatory verbosity and log format, may be empty
## Logger setup
- Convert config list to list of loggers
- If CMD provides log format, create logger and append to list
- If list is still empty, append logger with CMD verbosity and `plain` format
- Create logger which dispatches logs to each logger using slog::Duplicate
## Consequences
- User can log to multiple outputs, some may have different logging levels from others
- Config log format is always respected
- Sane default - same as in current solution
## ALSO log format
Log format should be upgraded from `LogFormat`
- `plain` (implicit written to stderr)
- `json` (implicitly written to stderr)

to `LogOutput`
- `stderr` (same as `plain`)
- `stderr_json` (same as `json`)
- `syslog`

**Comment by gufmar** [comment]
From a SysOps point of view and considering multi proxy-node setups it would be very useful to have syslog as an option.

As all the GRPC communication is stateless and happening to multiple neighbors in parallel I wonder if and how the logfiles can be written in order to improve readability. 
If there is a node-internal "session ID" valid for a certain task (who is a sequence of multiple stateless messages) it would be useful to have it in the logs. If there is no such internal ID, not even a sequential number counted up by the logger process, I have seen loggers in the past who sum up all messages for one process in the RAM and write it to the logfile at once (one start time, + x seconds since ...)

Also please consider to log all startup parameters only once (at startup) and don't repeat redundant information in many lines. Eg `allow_accounts: true, consensus-version: bft`

**Comment by CodeSandwich** [comment]
@gufmar I think that this fits another issue, which also is in progress: https://github.com/input-output-hk/jormungandr/issues/350

**Comment by mzabaluev** [comment]
I personally think `slog::Duplicate` and outputs with separate levels is overkill and it's a job for infrastructure to arrange any elaborate logging outside of the daemon if necessary.
On a single development host, it's enough if the log backend (e.g. journald) preserves the severity level, then the user can filter messages in the journal database or set up live diversions.
Buf if it's a "natural" thing to configure multiple backends in a non-exclusive fashion and trivial to do with slog, go for it.

**Comment by mzabaluev** [comment]
@gufmar We can generate per-task IDs if necessary, it seems easy enough to do with slog. Also, each node has an established ID for the p2p network protocol, and the log contexts in the network task are going to use that.

**Comment by NicolasDP** [comment]
Closing this issue as it has done its course. The duplication of outputs is indeed interesting but not necessary. @CodeSandwich can you add/details this in another issue so we can keep track of this and continue the discussion there?

**Comment by CodeSandwich** [comment]
Distilled multi-logger issue: https://github.com/input-output-hk/jormungandr/issues/392
